front/pages/api/v1/w/[wId]/assistant/agent_configurations.ts

```diff
@@ -53,20 +53,25 @@ async function handler(
     return apiError(req, res, keyRes.error);
   }
 
-  const { workspaceAuth, keyAuth } = await Authenticator.fromKey(
-    keyRes.value,
-    req.query.wId as string
-  );
-
-  if (
-    !workspaceAuth.isBuilder() ||
-    keyAuth.getNonNullableWorkspace().sId !== req.query.wId
-  ) {
+  const { wId } = req.query;
+  if (typeof wId !== "string") {
     return apiError(req, res, {
       status_code: 400,
       api_error: {
         type: "invalid_request_error",
-        message: "The Assistant API is only available on your own workspace.",
+        message: "Workspace ID not found",
+      },
+    });
+  }
+
+  const { workspaceAuth } = await Authenticator.fromKey(keyRes.value, wId);
+
+  if (!workspaceAuth || !workspaceAuth.isBuilder()) {
+    return apiError(req, res, {
+      status_code: 401,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Not enough permissions.",
```

Comment:
We usually avoid leaking information.
```suggestion
    return apiError(req, res, {
      status_code: 404,
      api_error: {
        type: "workspace_not_found",
        message: "The workspace you requested was not found.",
```

Reply:
Sounds weird to just drop a fake error - should we just say 'invalid request' then?

Reply:
It's not faking an error it's abstracting that a workspace exist if you don't have access. It's called `Security through Obscurity`.